pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -281,11 +284,11 @@ func xfrmDeleteConflictingState(states []netlink.XfrmState, new *netlink.XfrmSta
 					logfields.DestinationIP: s.Dst,
 				}).Info("Removed a conflicting XFRM state")
 			} else {
-				return deletedSomething, fmt.Errorf("Failed to remove a conflicting XFRM state: %w", err)
+				errs = errors.Join(errs, fmt.Errorf("Failed to remove a conflicting XFRM state: %w", err))
 			}
 		}
 	}
-	return deletedSomething, nil
+	return deletedSomething, errs
```

Comment:
I think what might be a better alternative here is to aggregate the msg and de-dup them a bit because then the error string might get too long given we are at a "leaf" function in the reconciliation loop. 

This can be done by counting the number of failures out of the number of states to delete, i.e. "5/10 failed", and keeping a set of the unique underlying errors that occurred. Given that we are interacting with netlink here, the error returned from those calls are going to be syscall-style errno error types such as EINVAL, ENOMEM, etc.

I would do something like this pseudocode:

```go
var (
	deletedSomething bool
	errorTypes       set.Set[error]
	failed           int
)
...
if err != nil {
	failed++
	errorTypes.Insert(err)
}
...
if failed > 0 {
	return deletedSomething, fmt.Errorf("failed to remove a conflicting XFRM states (%d/%d): %w", failed, len(states), errorTypes.Get()...)
}     
```

Reply:
Thanks Chris! Good point!! I'll lasso in something along those lines...